test/framework/clusterctl/client.go

```diff
@@ -137,6 +195,36 @@ func ConfigCluster(ctx context.Context, input ConfigClusterInput) []byte {
 	return yaml
 }
 
+// ConfigClusterWithBinary uses clusterctl binary to run config cluster.
+func ConfigClusterWithBinary(_ context.Context, clusterctlBinaryPath string, input ConfigClusterInput) []byte {
+	log.Logf("clusterctl config cluster %s --infrastructure %s --kubernetes-version %s --control-plane-machine-count %d --worker-machine-count %d --flavor %s",
+		input.ClusterName,
+		valueOrDefault(input.InfrastructureProvider),
+		input.KubernetesVersion,
+		*input.ControlPlaneMachineCount,
+		*input.WorkerMachineCount,
+		valueOrDefault(input.Flavor),
+	)
+
+	cmd := exec.Command(clusterctlBinaryPath, "config", "cluster",
+		input.ClusterName,
+		"--infrastructure", input.InfrastructureProvider,
+		"--kubernetes-version", input.InfrastructureProvider,
```

Comment:
```suggestion
		"--kubernetes-version", input.KubernetesVersion,
```
Hm, I'm curious why this didn't lead to errors in the test.